### PR TITLE
feat: list and delete extensions command

### PR DESCRIPTION
**Pull request by fredmaggiowski**
## What this PR is for?

Implement the extension **list** and **delete** (see #200 for specific changes) commands as proposed in #198 

**Which issue(s) this PR fixes:**

- Implements a part of #198 

Screenshot 2024-06-20 alle 11 33 22

### Notes

- This PR introduces a Table struct to abstract the table writer instance, if we like it I will change all table usages in a subsequent PR

### CODE REVIEW

File: internal/cmd/extensions/client.go [hunk 1]
```diff
@@ -65,7 +67,7 @@ func (e *E11yClient) Delete(ctx context.Context, companyID string, extensionID s
 		return fmt.Errorf("error executing request: %w", err)
 	}
 
-	if resp.StatusCode() > 399 {
+	if resp.StatusCode() > minimalSuccessStatusCode {
```
Comment (JGiola) [inline internal/cmd/extensions/client.go:70]: we can use http consts
```suggestion
	if resp.StatusCode() >= http.StatusBadRequest {
```

File: internal/cmd/extensions/delete.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+	"github.com/mia-platform/miactl/internal/client"
+	"github.com/mia-platform/miactl/internal/clioptions"
+
+	"github.com/spf13/cobra"
+)
+
+func addExtensionIDRequiredFlag(options *clioptions.CLIOptions, cmd *cobra.Command) {
+	flags := cmd.Flags()
+	flags.StringVarP(&options.EntityID, "extension-id", "i", "", "the extension id that should be deleted")
```
Comment (JGiola) [inline internal/cmd/extensions/delete.go:29]: I don't think that `i` is a meaningful abbreviation for `extension-id` can we avoid to set it?

File: internal/table/table.go
Comment (JGiola) [inline internal/table/table.go:1]: I'm not sell on adding this abstraction in this PR, particularly adding a whole new test library for strict string comparison.

Can you strip out this addition from the PR?

**Review by JGiola — CHANGES_REQUESTED** [review]

File: internal/cmd/extensions/delete.go (hunk 2 above)
Comment (fredmaggiowski) [reply internal/cmd/extensions/delete.go:29]: I've followed the abbreviation use for the marketplace item id but ok, i'm removing it

File: internal/cmd/extensions/client.go (hunk 1 above)
Comment (fredmaggiowski) [reply internal/cmd/extensions/client.go:70]: makes sense lol

File: internal/table/table.go
Comment (fredmaggiowski) [reply internal/table/table.go:1]: I agree with the uselessness of the new lib and I'm ok with stripping it, replicating the table dependency and configuration throughout the codebase however feels like a huge antipattern; i'd keep it. If you agree I can write a test utility to make assertion on tables which tbh are currently implemented in a pretty naive and fragile way

Comment (fredmaggiowski) [reply internal/table/table.go:1]: i've removed the lib and the abstraction; i'm opening an issue and a differentr pr for standardizing the pronter interface

**Review by JGiola — APPROVED** [review]

**Review by JGiola — CHANGES_REQUESTED** [review]
The PR is missing the addition to the CHANGELOG

**Review by JGiola — APPROVED** [review]

**Pull request merged**